### PR TITLE
feat(change_detection): do not register a change from switching from nul...

### PR DESCRIPTION
**Pull request by vsavkin**
...l to null

### CODE REVIEW

File: modules/change_detection/test/change_detection_spec.js [hunk 1]
```diff
@@ -291,8 +291,15 @@ export function main() {
               var dispatcher = c["dispatcher"];
 
               cd.detectChanges();
-              expect(dispatcher.log).toEqual(['a=null']);
-              dispatcher.clear();
+              expect(dispatcher.log).toEqual([]);
+            });
+
+            it("should register changes when switching from null to collection and back", () => {
```
Comment (rkirov) [inline modules/change_detection/test/change_detection_spec.js]: my pet peeve: use `'` or `"` consistently. Although, it looks like that ship has sailed in that file already, and I will have to do a global codebase cleanup.

File: modules/change_detection/src/array_changes.js [hunk 2]
```diff
@@ -52,6 +52,10 @@ export class ArrayChanges {
     return isListLikeIterable(obj);
   }
 
+  supportsObj(obj):boolean {
+    return ArrayChanges.supports(obj);
```
Comment (vicb) [inline modules/change_detection/src/array_changes.js]: do we need to keep the static method ?

File: modules/change_detection/src/change_detection_util.js [hunk 3]
```diff
@@ -146,7 +150,13 @@ export class ChangeDetectionUtil {
   }
 
   static structuralCheck(self, context) {
-    if (isBlank(self) || self === uninitialized) {
+    if (_isBlank(self) && _isBlank(context)) {
```
Comment (vicb) [inline modules/change_detection/src/change_detection_util.js]: factorize `_isBlank(context)`

**Comment by vsavkin** [comment]
Merged via https://github.com/angular/angular/commit/cd5d8bee949fef2345d3a44eda18111f6419709d

**Pull request closed without merging**